docs/gaslessWithERC4337.md

```diff
@@ -0,0 +1,131 @@
+
+### Difference between Gasless feature and ERC-4337
+
+For those who are familiar with ERC-4337 proposal, you might already find entrypoint and paymaster contracts quite ring a bell. In fact, the gasless feature is intended to use a simplified version of ERC-4337 specs. 
+The main difference is that ERC-4337 is also an account abstraction focus, which is to make mainstream users get rid of private keys and mnemonic stuff. You might often hear this as "smart-contract wallet". 
+
+In a high-level explanation, 
+> ERC-4337 = seedless feature + gasless feature
+
+While in Godwoken, we only do the last half of it. We believe it is a better choice for Godwoken to abort the seedless idea in the current phase, due to many reasons:
+- ERC-4337 is still in the protocol design phase, it is not widely accepted by the community. It might take a while before it makes a real influence on crypto. However, we want the users and developers to enjoy the gasless feature today.
+- Godwoken is focused on infrastructure while ERC-4337 is apparently an application-level business. The way Godwoken chooses to implement the gasless feature is more like an optional API, it doesn’t conflicts with the application developers who might wish to pursue a smart wallet opportunity when ERC-4337 is ready in the future.
+- ERC-4337 is very flexible and brings more complexity thus making it unrealistic to safely use in the production stage right now.
+- There is a gap between smart contract wallets and key-based wallets. Most users today are still using key-based wallets.
+- Since Godwoken is built on the UTXO model of the CKB blockchain, there is a lot of space to explore account abstraction ideas, and it might not happen on the EVM-compatible level.
+
+The seedless vision of ERC-4337 introduces a lot of complex details and involves different ecosystem actors in order to make it work. To implement account abstraction, ERC-4337 makes the user sign `UserOperation` call data and sends them to a separate mem-pool, A special class of actor called bundlers (either block builders, or users that can send transactions to block builders through a bundle marketplace) package up a set of these objects into a transaction making a special calling to `Entrypoint`, and that transaction then gets included in a block.
+Godwoken inherent the `UserOperation` idea from ERC-4337 but exclude the complexity part of account abstraction. The gasless feature allows the user to sign a transaction with 0 gas price to call entrypoint contract with `UserOperation` call data. Since the user is still signing a transaction instead of a `UserOperation` call data, we don't need a separate mem-pool to handle different kinds of requests. Therefore we don't need the bundlers role either. the block_producer is naturally the bundler who packs the gasless transaction into a new block. 
```

Comment:
```suggestion
Godwoken inherent the `UserOperation` idea from ERC-4337 but exclude the complexity part of account abstraction. The gasless feature allows the user to sign a transaction with 0 gas price to call entrypoint contract with `UserOperation` call data. Since the user is still signing a transaction instead of a `UserOperation` call data, we don't need a separate mem-pool to handle different kinds of requests. Therefore we don't need the role of bundlers either. The block_producer is naturally the bundler who packs the gasless transaction into a new block. 
```